### PR TITLE
Remove horse from events during cooldown period

### PR DESCRIPTION
**Pull request by parkerdavis1**
Continuing on from PR https://github.com/opportunity-hack/evs/pull/40

Removes horse from events during cooldown period.

The events are displayed as list of links.
Screen Shot 2023-08-06 at 14 14 11

### CODE REVIEW

File: app/routes/admin+/_horses+/horses.edit.$horseId.tsx [hunk 1]
```diff
@@ -80,6 +93,51 @@ export async function action({ request, params }: DataFunctionArgs) {
...
+						horses: {
+							disconnect: { id: updatedHorse.id },
+						},
+					},
+				})
+			}
+			return json({
+				status: 'idle',
+				submission,
+				conflictEvents,
```
Comment (parkerdavis1) [inline app/routes/admin+/_horses+/horses.edit.$horseId.tsx:136]: Can you think of a better way of returning these events to be displayed on the page? Typescript is unhappy that the action is returning conflictEvents.

Comment (benhsm) [reply app/routes/admin+/_horses+/horses.edit.$horseId.tsx:136]: I don't know if this is the best way to do this, but if you add a `conflictEvents: [],` field to the other return statements, that should resolve this particular type problem.

**Comment by parkerdavis1** [comment]
Screenshot 2023-08-08 at 10 49 31 AM
I didn't change anything here but now there's a typescript error on status. I'm having trouble finding the fortitude to deal with TS.

**Comment by benhsm** [comment]
You just need to add an `as const` assertion to the json return value on line 137 so it can treat the status string as an instance of the enum, and an empty `conflictEvents` array to return cases that don't have `conflictEvents` so the data returned from the action has a consistent shape

**Comment by benhsm** [comment]
```diff
diff --git a/app/routes/admin+/_horses+/horses.edit.$horseId.tsx b/app/routes/admin+/_horses+/horses.edit.$horseId.tsx
index 057bedf..510b4a8 100644
--- a/app/routes/admin+/_horses+/horses.edit.$horseId.tsx
+++ b/app/routes/admin+/_horses+/horses.edit.$horseId.tsx
@@ -58,13 +58,14 @@ export async function action({ request, params }: DataFunctionArgs) {
 	})
 
 	if (submission.intent !== 'submit') {
-		return json({ status: 'idle', submission } as const)
+		return json({ status: 'idle', submission, conflictEvents: [] } as const)
 	}
 	if (!submission.value) {
 		return json(
 			{
 				status: 'error',
 				submission,
+				conflictEvents: [],
 			} as const,
 			{ status: 400 },
 		)
@@ -134,7 +135,7 @@ export async function action({ request, params }: DataFunctionArgs) {
 				status: 'idle',
 				submission,
 				conflictEvents,
-			})
+			} as const)
 		}
 	}
 ```
 
 Are things ready to merge otherwise?

**Pull request merged**